lesson07.md

```diff
@@ -0,0 +1,363 @@
+<!-- .slide: id="lesson7" -->
+
+# Basic Frontend - Spring 2021
+
+Lesson 7, Tuesday, 2021-03-23
+
+---
+
+### Intro to JS functions
+
+---
+
+### Code's ability
+
+The facts are represented as values
+
+```js
+“Carlo” // someone has this as a name
+
+23 // maybe someone’s age, or shoe size
+
+true // it is the case that …
+```
+
+---
+
+Programs get all so powerful to the degree that they can produce new information from existing data
+
+For producing new info, we need operators
+
+```js
+2 + 3
+
+canSpeakEnglish || canSpeakGerman
+```
+
+---
+
+Operators provided by JS are limited in number
+
+- Arithmetic
+- Assignment
+- Comparison
+- Logical
+- String
+
+---
+
+Also limited in how many input values they take
+
+Takes 2:
+
+```js
+value1 + value2
+
+variable = value
+
+value1 < value2
+
+value1 || value2
+```
+
+---
+
+Takes 1:
+
+```js
+!value
+
+-value
```

Comment:
is this intended or was it supposed to be `--value`?

Reply:
It was intended as the unary negation operator that inverts the sign of the number. However --variable could have been another example for making the point as well. Just I wanted to stick to values as much as possible in the examples.